src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -222,7 +223,7 @@ public int process(List<File> files) throws CheckstyleException {
 
         final List<File> targetFiles = files.stream()
                 .filter(file -> CommonUtil.matchesFileExtension(file, fileExtensions))
-                .collect(Collectors.toList());
+                .collect(ImmutableList.toImmutableList());
```

Comment:
Opened a PR to get some feedback as the `mvn clean install` failed locally. The `Immutable*` import was not allowed. Isn't it and should I change this? Or do we need to add this somewhere such that it is allowed?

Reply:
Example of what I mean: https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=18567&view=logs&j=e663805c-ee13-5a24-f979-d725e0346c8e&t=40ff6e43-c5d7-5753-f947-23012624b0c6&l=1975.

Reply:
Long time ago we decided to reduce dependency to guava, as there was was problems in some plugins. I even don't remember root reason.

Can we archive the same by jdk methods ?

If not possible, we can reconsider old decision on guava, but it will take a bit longer to accept, as I need to recheck all old problems to be sure other will not be affected.

Reply:
Hmm, interesting. Internally we indeed like Guava and use these methods (although if there is a good JDK alternative we prefer that). In the `CollectorMutability` check we have the [following code](https://github.com/PicnicSupermarket/error-prone-support/blob/a2f44f82f2c815450c4fc3ee8b35452c03d053f1/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/CollectorMutability.java#L61):
```
if (!ThirdPartyLibrary.GUAVA.isIntroductionAllowed(state)
```
This piece of code basically makes the check do nothing as it doesn't want to introduce new Guava dependencies. Maybe we can introduce a different collector that still clearly communicates the immutability. It would not be something we prefer internally but would make sense if the introduction of new Guava dependencies is not desired.

I was playing around and found the `Collectors.toUnmodifiableList()` alternative to `ImmutableList.toImmutableList()`. I pushed a commit with a change where I applied this. Curious to hear what you think.

If we like the suggestion we can consider updating Error Prone Support to have that as alternative. We are always open to discussing improvements to Error Prone Support such that our checks are more useful to others.

Tagging @Stephan202 to get his opinion on this change proposal.

Reply:
Yeah, I think that it makes sense to suggest `Collectors.toUnmodifiable{List,Map,Set}` when `!ThirdPartyLibrary.GUAVA.isIntroductionAllowed(state)`, instead of backing out early like we do now. (We should then add a guard to prevent suggesting those variants when the build targets JDK < 10.)